executables/collector/src/main/kotlin/de/cyface/collector/handler/MeasurementHandler.kt

```diff
@@ -0,0 +1,573 @@
+            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            ctx.fail(ENTITY_UNPARSABLE, e)
+        } catch (e: IllegalSession) {
+            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            ctx.fail(ENTITY_UNPARSABLE, e)
+        } catch (e: PayloadTooLarge) {
+            LOGGER.error(String.format("Response: 422, %s", e.message), e)
+            ctx.fail(ENTITY_UNPARSABLE, e)
+        } catch (e: SessionExpired) {
+            LOGGER.warn(String.format("Response: 404, %s", e.message), e)
+            ctx.response().setStatusCode(NOT_FOUND).end() // client sends a new pre-request for this upload
+        } catch (e: SkipUpload) {
+            LOGGER.debug(e.message, e)
+            remove(session) // client won't resume
+            ctx.fail(PreRequestHandler.PRECONDITION_FAILED, e)
+        } catch (e: RuntimeException) {
+            ctx.fail(e)
+        }
+    }
+
+    /**
+     * Handles the chunk upload request when there was no chunk file found on the server yet.
+     *
+     * @param ctx The Vertx `RoutingContext` used to write the response
+     * @param request the request to read the body from
+     * @param session the session which was passed with the request
+     * @param user the user which was authenticated to this request
+     * @param contentRange the content range information from the header
+     * @param metaData the metadata from the request header
+     * @param path `String` if the session contained a path to the chunk file or `null` otherwise.
+     */
+    private fun handleFirstChunkUpload(
+        ctx: RoutingContext, request: HttpServerRequest,
+        session: Session, user: User, contentRange: ContentRange, metaData: RequestMetaData,
+        path: String?
+    ) {
+        if (contentRange.fromIndex != "0") {
+            if (path == null) {
+                // I.e. the server received data in a previous request but now cannot find the `path` session value.
+                // Unsure when this can happen. Not accepting the data. Asking to restart upload (`404`).
+                LOGGER.warn(String.format("Response: 404, path is null and unexpected content range: %s", contentRange))
+            } else {
+                // Server received data in a previous request but the chunk file was probably cleaned by cleaner task.
+                // Can't return 308. Google API client lib throws `Preconditions` on subsequent chunk upload.
+                // This makes sense as the server reported before that bytes (>0) were received.
+                LOGGER.warn(String.format("Response: 404, Unexpected content range: %s", contentRange))
+            }
+            ctx.response().setStatusCode(NOT_FOUND).end() // client sends a new pre-request for this upload
+            return
+        }
+        acceptUpload(ctx, request, session, user, contentRange, metaData)
+    }
+    // Cache found, expecting upload to continue the cached file
```

Comment:
This comment got lost. Please find its previous home.

Reply:
Ah, probably at line 201?

Reply:
This is in exactly the same position as in the old Java file, namely above the documentation for the method `handleSubsequentChunkUpload`